core/src/test/kotlin/io/forestframework/core/http/routing/MediaTypeNegotiatingIntegrationTest.kt

```diff
@@ -0,0 +1,163 @@
+    @TestFactory
+    fun testProducesAccept(): Iterable<DynamicTest> = producesAcceptCases.map(this::testOne)
+
+    private fun testOne(case: ProducesAcceptCase) = DynamicTest.dynamicTest(case.description) {
+        runBlockingUnit {
+            val vertx = Vertx.vertx()
+            val port = startTestApplication(vertx, case.httpMethod, produces = case.produces)
+
+            val httpClient = vertx.createHttpClient()
+            httpClient.get(port, "/test", mapOf("Accept" to case.accept)).assertStatusCode(case.expectedStatusCode)
+        }
+    }
+
+    /**
+     * Creates a mock server, registers the routing dynamically, waits for the server startup, then returns the port
+     */
+    private fun startTestApplication(
+        vertx: Vertx,
+        httpMethod: HttpMethod = HttpMethod.GET,
+        produces: List<String> = listOf("*/*"),
+        consumes: List<String> = listOf("*/*")
+    ): Int {
+        val configProvider = ConfigProvider(HashMap(), HashMap())
+        val extensions = listOf(
+            BindFreePortExtension(),
+            object : Extension {
+                override fun beforeInjector(startupContext: StartupContext?) {
+                    startupContext!!.componentClasses.add(WebRequestHandlingModule::class.java)
+                }
+
+                override fun afterInjector(injector: Injector) {
+                    injector.getInstance(RoutingManager::class.java).getRouting(HANDLER).add(
+                        DefaultRouting(false,
+                            HANDLER,
+                            "/test",
+                            "",
+                            listOf(httpMethod),
+                            MediaTypeNegotiatingIntegrationTest::class.java.getMethod("dummy"),
+                            0,
+                            produces,
+                            consumes)
+                    )
+                }
+            },
+            HttpServerExtension()
+        )
+        val context = DefaultStartupContext(vertx, MediaTypeNegotiatingIntegrationTest::class.java, configProvider, extensions)
+        Forest.run(context)
+
+        return configProvider.getInstance("forest.http.port", Integer::class.java).toInt()
+    }
+
+    @TestFactory
+    fun testConsumesContentType() = consumesContentTypeCases.map(this::testOne)
+
+    private fun testOne(case: ConsumesContentTypeCase) = DynamicTest.dynamicTest(case.description) {
+        runBlockingUnit {
+            val vertx = Vertx.vertx()
+            val port = startTestApplication(vertx, case.httpMethod, consumes = case.consumes)
+
+            val httpClient = vertx.createHttpClient()
+            httpClient.get(port, "/test", mapOf("Content-Type" to case.contentType)).assertStatusCode(case.expectedStatusCode)
+        }
+    }
+
+    @TestFactory
+    fun testMethodNotAllowed() = methodNotAllowedCases.map(this::testTwo)
+
+    private fun testTwo(case: MethodNotAllowedCase) = DynamicTest.dynamicTest(case.description) {
```

Comment:
```suggestion
    private fun testOne(case: MethodNotAllowedCase) = DynamicTest.dynamicTest(case.description) {
```

`testOne` means "test one case", not "test1"

Reply:
'kay, got it.